code/weapon/shockwave.cpp

```diff
@@ -297,32 +296,34 @@ void shockwave_move(object *shockwave_objp, float frametime)
 			}
 		}
 
-		// only apply damage to a ship once from a shockwave
-		for ( i = 0; i < sw->num_objs_hit; i++ ) {
-			if ( objp->signature == sw->obj_sig_hitlist[i] ){
+		bool found_in_list = false;
+
+		// only apply damage to an object once from a shockwave
+		for (auto & comparison : sw->obj_sig_hitlist) {
+			if ( (objp->signature == comparison.first) && (objp->type == comparison.second) ){
+				found_in_list = true;
 				break;
 			}
 		}
 
-		if ( i < sw->num_objs_hit ){
+		if (found_in_list) {
 			continue;
 		}
 
 		if ( weapon_area_calc_damage(objp, &sw->pos, sw->inner_radius, sw->outer_radius, sw->blast, sw->damage, &blast, &damage, sw->radius) == -1 ){
 			continue;
 		}
 
-		// okay, we have damage applied, record the object signature so we don't repeatedly apply damage
-		Assert(sw->num_objs_hit < SW_MAX_OBJS_HIT);
-		if ( sw->num_objs_hit >= SW_MAX_OBJS_HIT) {
-			sw->num_objs_hit--;
-		}
-
 		weapon_info* wip = NULL;
+		
+		// okay, we have damage applied, record the object signature so we don't repeatedly apply damage 
+		// but only add non-ships to the list if the Game_settings flag is set
+		if (objp->type == OBJ_SHIP || Shockwaves_damage_all_obj_types_once){
+		sw->obj_sig_hitlist.emplace_back(objp->signature, objp->type);
```

Comment:
Does this need to be indented or is the indentation correct?

Reply:
Blarg